agrirouter-api-java-impl/src/main/java/com/dke/data/agrirouter/impl/messaging/rest/EndpointsListServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+import com.dke.data.agrirouter.api.service.parameters.PayloadParameters;
+import com.dke.data.agrirouter.api.service.parameters.SendMessageParameters;
+import com.dke.data.agrirouter.impl.EnvironmentalService;
+import com.dke.data.agrirouter.impl.messaging.encoding.EncodeMessageServiceImpl;
+import com.dke.data.agrirouter.impl.validation.ResponseValidator;
+import java.util.Collections;
+import java.util.UUID;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class EndpointsListServiceImpl extends EnvironmentalService
+    implements EndpointsListService, MessageSender, ResponseValidator {
+
+  private EncodeMessageService encodeMessageService;
+  public EndpointsListServiceImpl(Environment environment) {
+    super(environment);
+    encodeMessageService = new EncodeMessageServiceImpl();
+  }
+
+  @Override
+  public String send(EndpointsListParameters parameters) {
+
+    // there is no validation required
+    // parameters.validate();
+
+    EncodeMessageResponse encodedMessage = encodeMessage(parameters);
+
+    SendMessageParameters sendMessageParameters = new SendMessageParameters();
+    sendMessageParameters.onboardingResponse = parameters.onboardingResponse;
+    sendMessageParameters.setEncodedMessages(
+        Collections.singletonList(encodedMessage.getEncodedMessage()));
+    
+
+    sendMessage(sendMessageParameters);
+
+    return encodedMessage.getApplicationMessageID();
+  }
+
+  private EncodeMessageResponse encodeMessage(EndpointsListParameters parameters) {
+
+    String applicationMessageID = UUID.randomUUID().toString();
+
+    MessageHeaderParameters messageHeaderParameters = new MessageHeaderParameters();
+    messageHeaderParameters.setApplicationMessageId(applicationMessageID);
+    messageHeaderParameters.setApplicationMessageSeqNo(1);
+    if (parameters.getUnFilteredList()) {
+      messageHeaderParameters.technicalMessageType =
+          TechnicalMessageType.DKE_LIST_ENDPOINTS_UNFILTERED;
+
+    } else {
+      messageHeaderParameters.technicalMessageType = TechnicalMessageType.DKE_LIST_ENDPOINTS;
+    }
+    messageHeaderParameters.mode = Request.RequestEnvelope.Mode.DIRECT;
+
+    PayloadParameters payloadParameters = new PayloadParameters();
+    payloadParameters.setTypeUrl(Endpoints.ListEndpointsQuery.getDescriptor().getFullName());
+    payloadParameters.value = new EndpointsListMessageContentFactory().message(parameters);
+
+    String encodedMessage =
+        this.encodeMessageService.encode(messageHeaderParameters, payloadParameters);
+
+    return new EncodeMessageResponse(applicationMessageID, encodedMessage);
+  }
+
+  public String requestFullListFiltered(OnboardingResponse onboardingResponse) {
+    EndpointsListParameters endpointsListParameters = new EndpointsListParameters();
+    endpointsListParameters.direction = Endpoints.ListEndpointsQuery.Direction.SEND_RECEIVE;
+    //null marks, that there shall be no filtering for a specific message type
```

Comment:
This should be part of the documentation within the parameters class and not at this point, could be interessting for all using the class.

Reply:
Whyever this was reverted. O_o null is NOT VALID, "" is the correct one. TechnicalMessageType.NONE is required.